apps/aetx/src/aetx.erl

```diff
@@ -298,6 +298,30 @@ type_to_cb(channel_settle_tx)         -> aesc_settle_tx;
 type_to_cb(channel_snapshot_solo_tx)  -> aesc_snapshot_solo_tx;
 type_to_cb(channel_offchain_tx)       -> aesc_offchain_tx.
 
+type_to_binary(spend_tx)                  -> <<"SpendTx">>;
```

Comment:
This refactoring introduces a non-reversable operation. Before you could to atom_to_binary and binary_to_atom back if need be. Now you rename and have no easy way to get your atom back.

Hard to see whether this has consequences for other parts of the code.

Therefore, I propose to compute the name instead of hard code it and to have type_to_binary and binary_to_type as part of the code.

A simple way to get from _ notation to camel notation is (not super beautiful, but works):
unicode:characters_to_binary([ string:titlecase(W) || W <- string:split(atom_to_binary(TxType, utf8), "_", all)]).

Reply:
Alternative (slightly faster in execution) you add the reverse for each keyword. But that is error prone and hard to maintain.

The best thing would be if we generated code for this in swagger_endpoints and use that... but that would introduce new dependencies.

Reply:
I don't think the reverse function is needed. We don't specify the tx type when we create a tx e.g. using the `/debug/*` endpoints - each tx type has its own endpoint so the tx type is not needed and so we don't have to convert it to atom in epoch.

Reply:
Ok, then a couple of observations

1. What is see is that the exported function `aetx:tx_type/1` is not used outside of the scope of this module

2. This function `type_to_binary/1` is only used in the `tx_type/1` and nowhere else

3. `tx_type/1` is used locally only in `serialize_for_client/1` that represents the transaction for client cosumption.

Based on these three points, I'd suggest:

1. Removing `tx_type/1`

2. Renaming `tx_type/1` to a more suitable name, that represents that this is only the swagger definition name there (`tx_swagger_type/1` maybe?) and using it in `serialize_for_client/1`. If so - we probably don't need the reverse function, indeed.

Reply:
We're probably safe using just `aetx:specialize_type/1` and `aetx:specialize_callback/1` depending on the specific case and this has no impact on any of them whatsoever.